src/devices/thermostat.cpp

```diff
@@ -365,6 +379,9 @@ std::shared_ptr<Thermostat::HeatingCircuit> Thermostat::heating_circuit(std::sha
     if (set_typeids.size()) {
         toggle_fetch(set_typeids[hc_num - 1], toggle_);
     }
+    if (set2_typeids.size()) {
```

Comment:
at this point hc_num could be higher than 4 , we should check 
`if (hc_num <= set2_typeids.size()) {`